sound/soc/sof/amd/acp.h

```diff
@@ -90,10 +93,6 @@ struct dma_descriptor {
 
 /* Scratch memory structure for communication b/w host and dsp */
 struct  scratch_ipc_conf {
-	/* DSP mailbox */
-	u8 sof_out_box[512];
-	/* Host mailbox */
-	u8 sof_in_box[512];
```

Comment:
can you please explain how this change is handled in a backwards-compatible way?

How would this work with previously released SOF firmware?

Reply:
> can you please explain how this change is handled in a backwards-compatible way?
> 
> How would this work with previously released SOF firmware?

The Changes are handled for backward compatible way also as the default values are same as previous firmware.  If firmware is not sending any window information the default values set will be configured/used. If the firmware sends the window information, then new offsets will be configured.  This won't break any of the previous firmware and help to set the mailbox offsets dynamically based on the firmware loaded. This implementation is compatible to the mailbox offsets of the firmware loaded based on the window information shared by the firmware. This will even work for future programs if mailbox sizes are changed by firmware in future programs.

Reply:
what about the flags, e.g. u32 sof_host_msg_write;? You didn't introduce any offset for those, so are they in different locations between firmware and driver?

Reply:
Flags are part of DSP memory. Flags are also taken care based on the information passed by the firmware. The offset of the flags are adjusted by adding the mailbox offset.

Eg. unsigned int dsp_msg_write = sdev->debug_box.offset +
				     offsetof(struct scratch_ipc_conf, sof_dsp_msg_write);